src/lib/Authentication/ExportAuthenticator.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzRecommendationClient\Authentication;
+
+use eZ\Publish\Core\REST\Common\Exceptions\NotFoundException;
+use EzSystems\EzRecommendationClient\Config\CredentialsCheckerInterface;
+use EzSystems\EzRecommendationClient\Helper\FileSystemHelper;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Authenticator for export feature, mainly used for basic auth based authentication.
+ */
+class ExportAuthenticator implements FileAuthenticatorInterface
+{
+    private const PHP_AUTH_USER = 'PHP_AUTH_USER';
+    private const PHP_AUTH_PW = 'PHP_AUTH_PW';
+
+    /** @var \EzSystems\EzRecommendationClient\Config\ExportCredentialsChecker */
+    private $credentialsChecker;
+
+    /** @var \Symfony\Component\HttpFoundation\RequestStack */
+    private $requestStack;
+
+    /** @var \EzSystems\EzRecommendationClient\Helper\FileSystemHelper */
+    private $fileSystem;
+
+    /**
+     * @param \EzSystems\EzRecommendationClient\Config\ExportCredentialsChecker $credentialsChecker
```

Comment:
I think it should be `\EzSystems\EzRecommendationClient\Config\CredentialsCheckerInterface` instead.